doc/xrcvd/transceiver-monitor-hld.md

```diff
@@ -283,6 +282,61 @@ An explicit "error clear event" can be added if some vendor's platform supports
 
 On transceiver plug in or plug out events, the port error status will be cleared.
 
+#### 1.4.3 Port Mapping Information handling ####
+
+xcvrd depends on port mapping information to update transceiver information to DB. Port mapping information contains following data:
+
+- Logical port name list. E.g. ["Ethernet0", "Ethernet4" ...]
+- Logical port name to physical port index mapping. E.g. {"Ethernet0": 1}
+- Physical port index to logical port name mapping. E.g. {1: "Ethernet0"}
+- Logical port name to ASIC ID mapping. This is useful for multi ASIC platforms.
+
+Currently, xcvrd assumes that port mapping information is never changed, so it always read static port mapping information from platform.json/port_config.ini and save it to a global data structure. However, things changed since dynamic port breakout feature introduced. Port can be added/created on the fly, xcvrd cannot update transceiver information, DOM information and transceiver status information without knowing the ports change. This causes data in state db not aligned with config db. To address this issue, xcvrd should subscribe CONFIG_DB PORT table change and update port mapping information accordingly.
+
+- Main process need not subscribe port configuration change.
+- State machine process and DOM sensor update thread subscribe port configuration change and update local port mapping accordingly.
+
+Port change event contains following data:
+
+- ASIC index which indicates the DB namespace.
+- Logical port name. Get from the key of PORT table. E.g, for key "PORT|Ethernet0", the logical port name is "Ethernet0".
+- Physic port index. Get from "index" field of PORT table.
+- Event type. Can be "Add" or "Remove".
+
+As port mapping information might be updated during runtime, the global port mapping information cannot be shared properly among main process, state machine process and DOM sensor update thread. A possible solution is to use share memory between different processes, but it will introduce process level lock to many places which is hard to maintain. So, a simple solution is to store local port mapping information in main process, state machine process and DOM sensor update thread and update them according to port configuration change. In this case, no explicit lock is needed and we can keep the logic as simple as it is. Of course it takes more memory, but port mapping information would be very small which should not cause any memory issue.
+
+##### 1.4.3.1 Subscribe CONFIG_DB PORT table change #####
+
+SONiC has implemented a way to "select" data changes from redis. xcvrd should reuse this "select" infrastructure to listen CONFIG_DB PORT table change. The workflow is like:
+
+1. For each ASIC namespace, create a selectable object which point to CONFIG_DB PORT table

```

Comment:
Here [https://github.com/Azure/sonic-platform-daemons/pull/212](https://github.com/Azure/sonic-platform-daemons/pull/212) mentioned only 3 cases? Which is correct?